### PR TITLE
Regex matching for some watcher commands

### PR DESCRIPTION
**Pull request by max0d41**
Wildcard watcher matching with the start/stop/restart commands is a good star. To handle two or more watcher "groups" it is not enough. I implemented regex matching in the form
`circusctl restart /^foo-(foo|bar)$/"`
The leading and ending slashes tells circus that this is a regex.

Would be nice to see this in the next release.

### CODE REVIEW

**Comment by k4nar** [comment]
It's a nice feature, however I wonder if it doesn't introduce too much complexity.

**Comment by max0d41** [comment]
Due to the non-blocking nature of circus it is not very comfortable to control more than one worker at once. Wildcards was not a good idea since it is not flexible enough, regex is much better.
Another solution I have in mind is to allow more than one argument to these commands (combined with the already implemented wildcards).
`circusctl restart foo-* bar-app`

What do you think?

**Comment by douardda** [comment]
Allowing more than one argument is a good idea, but is unrelated to the current PR.

Concerning the regex matching, I would prefer a more explicit approach of selecting the pattern syntax (glob vs regex), for example by using a command option like:

```
circusctl restart --regex foo-(foo|bar)
# or
circusctl restart --match=regex foo-(foo|bar)

```
Note that I've removed the ^ and $ on purpose (they should be added in the code)

David

**Comment by k4nar** [comment]
I kinda like the `--match` option. We could have three values here, `simple`, `glob` and `regex`. This way, even if someone would want to name its watcher with a *, it wouldn't be an issue.

**Comment by max0d41** [comment]
I think too the `--match` parameter is the cleanest implementation. Please test my last commit to see if everything is working as expected.

File: circus/commands/restart.py [hunk 1]
```diff
@@ -13,10 +13,23 @@ def execute_watcher_start_stop_restart(arbiter, props, watcher_function_name,
...
     """
     if 'name' in props:
-        name = re.compile(fnmatch.translate(props['name']))
-        watchers = [watcher
-                    for watcher in arbiter.iter_watchers()
-                    if name.match(watcher.name.lower())]
+        match = props.get('match', 'simple')
+        if match == 'simple':
+            try:
+                watchers = [arbiter.get_watcher(props['name'])]
```
Comment (k4nar) [inline circus/commands/restart.py]: You should use `self._get_watcher` instead, and remove the try/except.

**Comment by k4nar** [comment]
Should we have `simple` or `glob` as default? I think `glob` might be a good option, as it mimics the shell behavior.

**Comment by k4nar** [comment]
Also, could you remove those file mode changes?

**Comment by max0d41** [comment]
Here you go. I did a rebase to keep everything clean.
The +x file modes are useless so I removed them in an extra commit. Cherry pick if you want to keep them.

**Comment by k4nar** [comment]
Awesome, thanks.

However, you have some PEP8 errors: https://travis-ci.org/circus-tent/circus/jobs/64465731

**Comment by max0d41** [comment]
The PEP8 errors are fixed.

**Comment by k4nar** [comment]
@max0d41: You still have one error https://travis-ci.org/circus-tent/circus/jobs/64577473 :)

**Comment by max0d41** [comment]
Next time I will wait until travis is done, but not now :D
I think this is the last rebase I have to push.

**Comment by k4nar** [comment]
Everything's fine now :) .

Thank you very much for your patience!

**Pull request merged**